### PR TITLE
Release 3.3.0

### PR DESCRIPTION
**Pull request by lucacome**
### Proposed changes

Docs changes for `v3.3.0`

### CODE REVIEW

File: docs/content/technical-specifications.md [hunk 1]
```diff
@@ -65,16 +66,16 @@ NGINX Plus images are available through the F5 Container registry `private-regis
...
+|Debian-based image | ``debian:12-slim`` | NGINX Plus JavaScript and OpenTracing modules, OpenTracing tracers for Jaeger, Zipkin and Datadog | `nginx-ic/nginx-plus-ingress:3.3.0` | arm64, amd64 |
+|Debian-based image with NGINX App Protect WAF | ``debian:11-slim`` | NGINX App Protect WAF, NGINX Plus JavaScript and OpenTracing modules, OpenTracing tracers for Jaeger, Zipkin and Datadog | `nginx-ic-nap/nginx-plus-ingress:3.3.0` | amd64 |
+|Debian-based image with NGINX App Protect DoS | ``debian:11-slim`` | NGINX App Protect DoS, NGINX Plus JavaScript and OpenTracing modules, OpenTracing tracers for Jaeger, Zipkin and Datadog | `nginx-ic-dos/nginx-plus-ingress:3.3.0` | amd64 |
+|Debian-based image with NGINX App Protect WAF and DoS | ``debian:11-slim`` | NGINX App Protect WAF and DoS, NGINX Plus JavaScript and OpenTracing modules, OpenTracing tracers for Jaeger, Zipkin and Datadog | `nginx-ic-nap-dos/nginx-plus-ingress:3.3.0` | amd64 |
+|Ubi-based image | ``redhat/ubi9-minimal`` | NGINX Plus JavaScript module | `nginx-ic/nginx-plus-ingress:3.3.0-ubi` | arm64, amd64, s390x |
```
Comment (brianehlert) [inline docs/content/technical-specifications.md:75]: There should also be a ubi8-minimal option here.  Do we have a naming / label pattern to support both?

Comment (lucacome) [reply docs/content/technical-specifications.md:75]: we don't have a ubi8 based image yet

**Review by brianehlert — CHANGES_REQUESTED** [review]

File: docs/content/releases.md [hunk 2]
```diff
@@ -6,32 +6,69 @@ doctypes: ["concept"]
 toc: true
 docs: "DOCS-616"
 ---
+<hr>
 
-## NGINX Ingress Controller 3.2.1
+## 3.3.0
+
+26 Sep 2023
+
```
[the remark this answers is not in the document]
Comment (brianehlert) [reply docs/content/releases.md:14]: What is the current method for a "featured" or "highlights" section?
Seems like we used to do one but fell out of practice the past few releases.

[the remark this answers is not in the document]
Comment (lucacome) [reply docs/content/releases.md:14]: yeah we add an overview sometimes, just wanted to confirm that you wanted me to add it.

**Review by shaun-nx — APPROVED** [review]

**Pull request merged**